### PR TITLE
Nucleo F303K8 build

### PR DESCRIPTION
**Pull request by ducky64**
Adds build environments and a main build target for the Nucleo F303K8. All existing files using the LPC1549 are still built to the LPC1549, the Nucleo source files are in the `src_stm32f303` directory - so the current lab documents don't change. 

I've left the PwmOut and DigitalIn commented out because those will be replaced with the actual pinouts to the breadboard circuit.

A future PR will update the lab documents to use the Nucleo F303K8, as well as refactor the lab structure so the first labs don't need external hardware (only rely on a LED and serial terminal).

@dragonlock2 if you have a Nucleo F303K8, can you test if this works? Run `scons flash-stm32f303` with the Nucleo connected. It should blink the green LED at ~5Hz.

### CODE REVIEW

File: SConscript-env-stm32f303 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+mbed_sources = env.GetMbedSources(mbed_paths)
+mbed_lib = env_mbed.StaticLibrary('mbed', mbed_sources)
+env.Prepend(LIBS=mbed_lib)
+
+# Build mbed RTOS as a static library
+rtos_paths = [
+  Dir('mbed/rtos'),
+  Dir('mbed/rtos/rtx/TARGET_CORTEX_M'),
+  Dir('mbed/rtos/rtx/TARGET_CORTEX_M/TARGET_M3'),
+  Dir('mbed/rtos/rtx/TARGET_CORTEX_M/TARGET_M3/TOOLCHAIN_GCC'),
```
Comment (dragonlock2) [inline SConscript-env-stm32f303:27]: f303k8 is a Cortex-M4 so this should be TARGET_RTOS_M4_M7. Didn't make a difference when I tried it though so probably doesn't matter?

**Comment by dragonlock2** [comment]
Everything looks good. I don't have a Nucleo F303K8, but I changed two things and got it compiling/flashing for a F103 and also tried using Thread and everything seems to work.

File: SConscript-env-stm32f303 (hunk 1 above)
Comment (ducky64) [reply SConscript-env-stm32f303]: Whoops, fixed now.

**Comment by ducky64** [comment]
Cool. I also added a test of the RGB LED, basically a direct copy of the lab1.5 solution for the lpc1549 with pindefs changed.

**Pull request merged**